### PR TITLE
[WIP]feature/Chioma/Log_Out_User

### PR DESCRIPTION
**Pull request by Simpcyclassy**
Work in Progress

### CODE REVIEW

File: src/components/app/reducers.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const initialState = {
+    loggedIn: false,
+    user: { ... }
+};
+
+export default ( state = { ...initialState }, action) => {
+    switch (action.type) {
+        case SET_COOKIE:    {
+            const expiration = new Date().getTime()
```
Comment (enyeInc) [inline src/components/app/reducers.js]: How long do you want to cookie to live before expiring?

File: src/components/app/actions.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Triggers request to remove cookie
+ *
+ * @function
+ * @return {Object} The {@link actionTypes.REMOVE_COOKIE REMOVE_COOKIE} action.
+ */
+export const removeCookie = () => ({
+    type: REMOVE_COOKIE,
+});
```
Comment (enyeInc) [inline src/components/app/actions.js]: @Simpcyclassy Where are these actions being used?

File: package.json [hunk 3]
```diff
@@ -66,6 +66,7 @@
     "@babel/preset-react": "^7.0.0",
     "@babel/preset-stage-2": "^7.0.0",
     "@babel/runtime": "^7.1.2",
+    "babel-eslint": "^10.0.1",
```
Comment (Way2nnadi) [inline package.json]: why did you add this?

File: src/components/signin/components/Signin.js [hunk 4]
```diff
@@ -52,6 +53,7 @@ class Signin extends React.Component {
 
         if(signinStatus){
             message.success(signinMessage, 5);
+            setCookie();
```
Comment (Way2nnadi) [inline src/components/signin/components/Signin.js]: This function will not be picked up by redux. You need to grab the action from the props of the component because that action has been bound to dispatch (dispatch is how components alert redux that an action has fired)

File: src/components/signin/components/Signin.js [hunk 5]
```diff
@@ -93,7 +95,7 @@ const mapStateToProps = state => ({
 });
 
 const mapDispatchToProps = dispatch => ({
-    actions: bindActionCreators(signinActions, dispatch),
+    actions: bindActionCreators(Object.assign({}, signinActions, setCookie), dispatch),
```
Comment (Way2nnadi) [inline src/components/signin/components/Signin.js]: i prefer that you use es7 object spread operator here
```
bindActionCreators({ ... signinActions, setCookie }, dispatch)
```

File: src/components/app/components/Navbar.js [hunk 6]
```diff
@@ -21,8 +24,12 @@ const { ContactUs } = home.components;
 const { Signin } = signin.components;
 const { Signup } = signup.components;
 
+const onClick = ({ key }) => {
+    key === {SIGN_OUT}? this.logout : ""
+};
+
```
Comment (enyeInc) [inline src/components/app/components/Navbar.js]: This seems wrong. did you test this code?

File: src/components/app/reducers.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    user: "",
+};
+
+export default ( state = { ...initialState }, action) => {
+    switch (action.type) {
+    case SET_COOKIE:    {
+        const time = 12;
+        const expiration = new Date(new Date().getTime()+ time * 3600 * 1000);
+        const token = action.data.token || 'testtoken';
+        browser.cookies.set(token, expiration);
```
Comment (enyeInc) [inline src/components/app/reducers.js]: why are you still using `browser` here. Did we decided that you would be using `cookies-js`?

File: package.json [hunk 8]
```diff
@@ -32,6 +32,7 @@
     "bcrypt-nodejs": "0.0.3",
     "body-parser": "^1.18.3",
     "compression": "^1.7.3",
+    "cookies-js": "^1.2.3",
```
Comment (enyeInc) [inline package.json]: where are you using this module?
